src/intersection.js

```diff
@@ -23,5 +23,7 @@ var uniq = require('./uniq');
  *      R.intersection([1,2,3,4], [7,6,5,4,3]); //=> [4, 3]
  */
 module.exports = _curry2(function intersection(list1, list2) {
-  return uniq(_filter(flip(_contains)(list1), list2));
+  var lookupList = list1.length > list2.length ? list1 : list2;
+  var filteredList = lookupList == list1 ? list2 : list1;
```

Comment:
@asaf-romano:
>> Ahh, for the days when we can count on destructuring...

> it's just a matter of integrating with babel ;)

Yes, if we want this in our source:

```js
var _slicedToArray = (function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; })();

var _ref = list1.length > list2.length ? [list1, list2] : [list2, list1];

var _ref2 = _slicedToArray(_ref, 2);

var lookupList = _ref2[0];
var filteredList = _ref2[1];
```
(http://bit.ly/1U2w2pB)  I have no idea why it needs that `_slicedToArray` monstrosity here.  It's _creating_ a clearly iterable instance the line above.